### PR TITLE
doc: add ceph fs volumes and subvolumes documentation

### PR DESCRIPTION
**Pull request by ajarr**
Fixes: https://tracker.ceph.com/issues/40689
Signed-off-by: Ramana Raja <rraja@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by batrick** [comment]
jenkins render docs

File: doc/cephfs/fs-volumes.rst [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    $ ceph fs volume create <vol_name>
+
+This creates a CephFS file sytem and its data and metadata pools. It also tries
+to create MDSes for the filesytem using the enabled ceph-mgr orchestrator
+module  (see :doc:`/mgr/orchestrator_cli`) , e.g., rook.
+
+Remove a volume using::
+
+    $ ceph fs volume rm <vol_name>
```
Comment (batrick) [inline doc/cephfs/fs-volumes.rst]: Are there any protections built into this command like `--yes-i-really-mean-it`?

File: doc/cephfs/fs-volumes.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+The ceph-mgr volumes module implements the following file system export
+abstactions:
+
+* FS volumes, an abstraction for CephFS file systems
+
+* FS subvolumes, an abstraction for independent CephFS directory trees
+
+* FS subvolume groups, an abstraction for a directory level higher than FS
+  subvolumes
```
Comment (batrick) [inline doc/cephfs/fs-volumes.rst]: ... to effect policies across a set of subvolumes like file layouts.

File: doc/cephfs/fs-volumes.rst [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    $ ceph fs subvolumegroup snapshot create <vol_name> <group_name> <snap_name>
+
+
+Remove a snapshot of a subvolume group using::
+
+    $ ceph fs subvolumegroup snapshot rm <vol_name> <group_name> <snap_name> [--force]
+
+Using the '--force' flag allows the command to succeed that would otherwise
+fail if the snapshot did not exist.
```
Comment (batrick) [inline doc/cephfs/fs-volumes.rst]: How to list subvolume group snapshots?

Should note that this implicitly snapshots all subvolumes under the subvolume group.

File: doc/cephfs/fs-volumes.rst (hunk 1 above)
Comment (ajarr) [reply doc/cephfs/fs-volumes.rst]: Not yet. Created a tracker ticket for now, https://tracker.ceph.com/issues/41841.

@batrick Do you want me to skip documenting FS volume commands until this is fixed?

File: doc/cephfs/fs-volumes.rst (hunk 3 above)
Comment (ajarr) [reply doc/cephfs/fs-volumes.rst]: > How to list subvolume group snapshots?
> 

We don't have list commands yet. Created a tracker ticket,
https://tracker.ceph.com/issues/41842

> Should note that this implicitly snapshots all subvolumes under the subvolume group.

Yes. I will fix this.

File: doc/cephfs/fs-volumes.rst (hunk 2 above)
Comment (ajarr) [reply doc/cephfs/fs-volumes.rst]: Done

File: doc/cephfs/fs-volumes.rst (hunk 1 above)
Comment (batrick) [reply doc/cephfs/fs-volumes.rst]: No, @joscollin can update the documentation as part of his PR.

**Comment by batrick** [comment]
Needs rebased.

**Comment by ajarr** [comment]
> Ready to merge IMO once rebased.

Rebased

**Comment by ajarr** [comment]
@vshankar can you review and merge?

**Pull request merged**